### PR TITLE
Hotfix/display email registration

### PR DESCRIPTION
**Pull request by RomanKuk**
fix bug of setting token before reg/login
fix bug of display email (need to change firebase settings 1 acc - 1 email, drop db and delete firebase users)
fix bug of linking account
link buttons are not in the correct place
link button can be disabled only after refreshing page (when user resolve set current user)

### CODE REVIEW

File: backend/buildeR.BLL/Services/UserService.cs [hunk 1]
```diff
@@ -121,5 +121,32 @@ public async Task<bool> ValidateUsername(ValidateUserDTO user)
...
+        public async Task<UserDTO> LinkProvider(LinkProviderDTO userLink)
+        {
+            var isUserExist = await _context.Users.AnyAsync(u => u.Id == userLink.UserId);
+
+            if (isUserExist)
+            {
+                var userSN = new NewUserSocialNetworkDTO()
+                {
+                    UId = userLink.UId,
+                    SocialNetworkId = (int)userLink.ProviderId + 1,
```
Comment (VoBilyk) [inline backend/buildeR.BLL/Services/UserService.cs:134]: What are you doing here?

File: frontend/src/app/core/services/authentication.service.ts [hunk 2]
```diff
@@ -41,9 +46,18 @@ export class AuthenticationService {
...
-      user => {
-        this.currentUser = user;
-        this.router.navigate(['/portal']);
+      userResult => {
+        this.angularAuth.authState
+          .pipe(takeUntil(this.unsubscribe$))
+          .subscribe((user) => {
+            this.configureAuthState(user);
+            if (user) {
+              if (user.uid === userResult.userSocialNetworks[0].uId) {
```
Comment (VoBilyk) [inline frontend/src/app/core/services/authentication.service.ts:55]: Why don't you combine those conditions?

File: frontend/src/app/core/services/firebase-sign-in.service.ts [hunk 3]
```diff
@@ -1,52 +1,158 @@
...
 import { AuthenticationService } from './authentication.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class FirebaseSignInService {
 
+  private unsubscribe$ = new Subject<void>();
```
Comment (VoBilyk) [inline frontend/src/app/core/services/firebase-sign-in.service.ts:19]: What are you doing with this? It's never emited

File: frontend/src/app/modules/user/user.component.ts [hunk 4]
```diff
@@ -12,34 +14,57 @@ import { TabRoute } from '@shared/models/tabs/tab-route';
...
+    this.fbr.linkWithGithub();
+  }
+
+  linkWithGoogle() {
+    this.fbr.linkWithGoogle();
+  }
+
+  isProviderAdded(provider: Providers) {
+    for (const item of this.currentUser.userSocialNetworks) {
+      if (item.socialNetworkId - 1 === Number(provider)) {
```
Comment (VoBilyk) [inline frontend/src/app/modules/user/user.component.ts:63]: You can change it Array.some(...). But still what do you do with item.socialNetworkId - 1

**Review by VoBilyk — COMMENTED** [review]
I see a few strange moments

**Comment by VoBilyk** [comment]
I'll approve it to be able to play with it, but don't forget to fix the remarks

File: backend/buildeR.BLL/Services/UserService.cs (hunk 1 above)
Comment (RomanKuk) [reply backend/buildeR.BLL/Services/UserService.cs:134]: Maybe, I had to change enum (starting from 1), now it starts from 0, that's why

File: frontend/src/app/core/services/authentication.service.ts (hunk 2 above)
Comment (RomanKuk) [reply frontend/src/app/core/services/authentication.service.ts:55]: Yeah, I forgot about that

File: frontend/src/app/core/services/firebase-sign-in.service.ts (hunk 3 above)
Comment (RomanKuk) [reply frontend/src/app/core/services/firebase-sign-in.service.ts:19]: I was trying different ways, and forgot about it, but it doesnt influence on the result

File: frontend/src/app/modules/user/user.component.ts (hunk 4 above)
Comment (RomanKuk) [reply frontend/src/app/modules/user/user.component.ts:63]: I take it into account

**Comment by RomanKuk** [comment]
> I'll approve it to be able to play with it, but don't forget to fix the remarks

Ok, I fix it for sure))

**Pull request merged**